framework/components/ecl-forms/ecl-forms-checkboxes/ecl-forms-checkboxes.twig

```diff
@@ -1,43 +1,51 @@
+{#
+  Accepts:
+    - "id" (string): checkbox group ID (mandatory)
+    - "name" (string): checkbox name (mandatory)
+    - "value" (string): checkbox value (optional, default: '')
+    - "label" (string): checkbox label (optional, default: '')
+    - "checked" (boolean): is checkbox checked? (optional, default: false)
+    - "is_disabled" (boolean): is disabled (optional, default: false)
+    - "has_error" (boolean): has error (optional, default: false)
+    - "extra_class" (string): extra CSS classes (optional, default: '')
+#}
+
 {% set id = id|default('default-id') %}
 {% set name = name|default('default-name') %}
 {% set checked = checked|default(false) %}
 
 {# Internal properties #}
 
 {% set _cssClass = 'ecl-checkbox' %}
-{% set _extraAttributes = '' %}
+{% set _extra_attributes = '' %}
 
 {# Internal logic - Process properties #}
 
-{% if hasError is defined and hasError == true %}
+{% if has_error is defined and has_error == true %}
   {% set _cssClass = _cssClass ~ ' ecl-checkbox--has-error' %}
 {% endif %}
 
-{% if extraClass is defined %}
-  {% set _cssClass = _cssClass ~ ' ' ~ extraClass %}
+{% if is_disabled is defined and is_disabled == true %}
+  {% set _cssClass = _cssClass ~ ' ecl-checkbox--is-disabled' %}
 {% endif %}
 
-{% if isDisabled is defined and isDisabled == true %}
-  {% set _extraAttributes = _extraAttributes ~ ' disabled' %}
+{% if extra_classes is defined %}
+  {% set _cssClass = _cssClass ~ ' ' ~ extra_classes %}
 {% endif %}
 
-{% if extraAttributes is defined %}
-  {% for attr in extraAttributes %}
-    {% set _extraAttributes = _extraAttributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~'"' %}
+{% if extra_attributes is defined %}
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~'"' %}
   {% endfor %}
 {% endif %}
 
 {# Print the result  #}
 
-<input
-  type="checkbox"
-  class="{{ _cssClass }}"
-  id="{{ id }}"
-  name="{{ name }}"
-  value="{{ value }}"
-  ckecked="{{ checked }}"
-  {{ _extraAttributes }}
->
-{% if label is defined %}
-<label class="ecl-checkbox__label" for="{{ id }}">{{ label }}</label>
-{% endif %}
+{% spaceless %}
+{% embed '@ec-europa/ecl-forms-labels' with { 'for': id, 'extra_classes': _cssClass, 'extra_attributes': _extra_attributes, _id: id, _value: value, _name: name, _label: label, _checked: checked, _is_disabled: is_disabled } only %}
+  {% block label %}
+    <input class="ecl-checkbox__input ecl-sr-only" type="checkbox" id="{{ _id }}" value="{{ _value }}" name="{{ _name }}"{{ _checked ? ' checked' : '' }}{{ (_is_disabled is defined and _is_disabled == true) ? ' disabled' : '' }} />
```

Comment:
Ok, I learned something new. Honestly, I still feel it's not very readable, but since it works, it's not a problem.